src/sources/amqp.rs

```diff
@@ -0,0 +1,565 @@
+use crate::{
+    amqp::AMQPConfig,
+    codecs::{Decoder, DecodingConfig},
+    config::{log_schema, Output, SourceConfig, SourceContext},
+    event::{BatchNotifier, BatchStatus},
+    internal_events::{
+        source::{AMQPEventError, AMQPEventReceived},
+        StreamClosedError,
+    },
+    serde::{bool_or_struct, default_decoding, default_framing_message_based},
+    shutdown::ShutdownSignal,
+    SourceSender,
+};
+use async_stream::stream;
+use bytes::Bytes;
+use chrono::{TimeZone, Utc};
+use codecs::decoding::{DeserializerConfig, FramingConfig};
+use futures::{FutureExt, StreamExt};
+use lapin::{acker::Acker, message::Delivery, Channel};
+use snafu::Snafu;
+use std::io::Cursor;
+use tokio_util::codec::FramedRead;
+use vector_common::finalizer::UnorderedFinalizer;
+use vector_config::configurable_component;
+use vector_core::{
+    config::{AcknowledgementsConfig, LogNamespace},
+    event::Event,
+};
+
+#[derive(Debug, Snafu)]
+enum BuildError {
+    #[snafu(display("Could not create AMQP consumer: {}", source))]
+    AMQPCreateError {
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+    #[snafu(display("Could not subscribe to AMQP queue: {}", source))]
+    AMQPSubscribeError { source: lapin::Error },
+}
+
+/// Configuration for the `amqp` source.
+#[configurable_component(source("amqp"))]
+#[derive(Clone, Debug, Derivative)]
+#[serde(deny_unknown_fields)]
+pub struct AMQPSourceConfig {
+    /// The queue.
```

Comment:
Is this the queue itself or the name of the queue ? :thinking: Just seeing the `String` type makes me a little puzzled that this would be the actual queue, it seems more like the queue name.

Reply:
The name of the queue.